reports-scheduler/src/main/kotlin/com/amazon/opendistroforelasticsearch/reportsscheduler/model/InContextReportCreateRequest.kt

```diff
@@ -117,9 +119,18 @@ internal class InContextReportCreateRequest : ActionRequest, ToXContentObject {
                 }
             }
         }
-        beginTime ?: throw IllegalArgumentException("$BEGIN_TIME_FIELD field absent")
-        endTime ?: throw IllegalArgumentException("$END_TIME_FIELD field absent")
-        status ?: throw IllegalArgumentException("$STATUS_FIELD field absent")
+        beginTime ?: run {
+            Metrics.getInstance().getNumericalMetric(MetricName.REPORT_FROM_DEFINITION_USER_ERROR).increment()
+            throw IllegalArgumentException("$BEGIN_TIME_FIELD field absent")
+        }
+        endTime ?: run {
+            Metrics.getInstance().getNumericalMetric(MetricName.REPORT_FROM_DEFINITION_USER_ERROR).increment()
+            throw IllegalArgumentException("$END_TIME_FIELD field absent")
+        }
+        status ?: run {
+            Metrics.getInstance().getNumericalMetric(MetricName.REPORT_FROM_DEFINITION_USER_ERROR).increment()
```

Comment:
can we have dimensions so that stats can have fine grained info

Reply:
I did not understand this. Can you provide an example?

Reply:
Will make this more fined grained fo easier debugging.

Reply:
Done.